### ISSUE TITLE
Lots of test suite errors of version 1.7 on i386

### ISSUE THREAD

**Issue by tillea** [samtools/bcftools#744]
Hi,
I've uploaded packages of version 1.7.  Unfortunately the build time tests under i386 (and other architectures) [failed](https://buildd.debian.org/status/fetch.php?pkg=bcftools&arch=i386&ver=1.7-1&stamp=1518882841&raw=0).  In the end you get:

    Number of tests:
        total   .. 795
        passed  .. 693
        failed  .. 102

I admit I have no idea how to fix this -version 1.6 was OK in this respect.
Kind regards, Andreas.

**Comment by ginggs** [comment]
I don't believe bcftools has ever built on i386 for Debian due to the use of signalling NaNs.
https://github.com/samtools/hts-specs/issues/145

On 32-bit ARM and MIPS architectures, we are getting a single new test failure in 1.7.

```
Number of tests:
    total   .. 795
    passed  .. 794
    failed  .. 1
```

```
test_vcf_query:
	/<<PKGBUILDDIR>>/bcftools query -f'[%POS  %SAMPLE  %AD\n]\n' -i'AD[:0]+AD[:1] > 12' /tmp/VDxYjli8aH/filter.5.vcf.gz
malloc(): memory corruption

	The outputs differ:
		/<<PKGBUILDDIR>>/test/query.54.out
		/<<PKGBUILDDIR>>/test/query.54.out.new
.. failed ...
```

**Comment by tillea** [comment]
I've got another hint that the issue is related to issue #389.

**Comment by tillea** [comment]
I've edited my original issue to make it different from #389.

**Comment by pd3** [comment]
I am not able to reproduce the problem on my platform and valgrind does not complain either. Very likely this is also related to #389
